test/yetibot/core/test/config.clj

```diff
@@ -25,12 +27,14 @@
          (fact "returns empty map when no valid YB ENV vars are present"
                yb-env-vars => (every-checker map? empty?))))
 
-(facts "about config-from-env-or-file"
-       (let [yb-vars (c/config-from-env-or-file {:yetibot {:a 1}}
-                                                {:yetibot-b-c 2})]
-         (fact "returns expected KV items in non-empty map with
-                valid (user defined) YB file cfgs and env cfgs"
-               yb-vars => (every-checker map? not-empty)
-               (:a yb-vars) => 1
-               (:b yb-vars) => (every-checker map? not-empty)
-               (get-in yb-vars [:b :c]) => 2)))
+(facts
+ "about config-from-env-or-file"
+ (fact
+  "returns expected KV items in non-empty map with valid (user defined)
+   YB file cfgs and env cfgs"
+  (c/config-from-env-or-file) => (every-checker map? not-empty)
+  (:a (c/config-from-env-or-file)) => 1
+  (:b (c/config-from-env-or-file)) => (every-checker map? not-empty)
+  (get-in (c/config-from-env-or-file) [:b :c]) => 2
+  (against-background (uc/load-edn! anything) => {:yetibot {:a 1}}
```

Comment:
have you figured out when to use `against-background` vs `provided`?

Reply:
maybe i am misinterpreting it -- but i was going off of the "a pool of facts that apply in many situations" statement as defined here :: https://github.com/marick/Midje/wiki/Background-prerequisites

so my logic is simply, i have multiple "situations" (multiple tests) and need to mock for .. so instead of having a `(provided)` for each situation -- i can just declare my "background" once and use it many times ..

when i had tried to provide a single `(provided)` for many "situations" (again, tests) -- i was getting this weird error (wish i saved it) talking about how the previous expression is not "suitable" for the `(provided)` context ..

if i am apply this incorrectly -- any guidance you can share is appreciated ..

Reply:
I see, this makes sense! I was previously a bit confused on the difference (even though I currently use them [here](https://github.com/yetibot/core/blob/6fa4584e20b7e8adede405a3492dbbfd2237d7f0/test/yetibot/core/test/util/command.clj#L20-L39)) but the section of the docs makes it pretty clear. Thanks 👍